samples/tutorials/hardwareInterfaces/i2c/BatteryChargerReading/batteryReadingComponent/batteryReading.c

```diff
@@ -0,0 +1,275 @@
+static void EnableI2cBus
+(
+    void
+)
+{
+    LE_DEBUG("Enabling TCA9546A I2C switch...");
+    const int i2cBus = 0;
+    const int i2cdev_fd = I2cAccessBusAddr(i2cBus, 0x71);
+    LE_FATAL_IF(i2cdev_fd == LE_FAULT, "failed to open i2cbus %d\n", i2cBus);
+
+    const uint8_t enableAllPorts = 0xff;
+    LE_FATAL_IF(i2c_smbus_write_byte(i2cdev_fd, enableAllPorts) == -1, "failed to write i2c data");
+}
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * This functions reads and prints all the registers on  an SMBUS write of a 1 byte register
+ *
+ */
+//--------------------------------------------------------------------------------------------------
+static void ReadBatterychargerRegister
+(
+    void
+)
+{
+// read all ten registers of Battery charger with i2c address 0x71
+    uint8_t RegisterReading;
+
+    SmbusReadReg(0, 0x6B, 0x00, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Input Source Control value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x01, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Power-On Configuration Register value is %d\n",RegisterReading);
+    
+    SmbusReadReg(0, 0x6B, 0x02, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Charge Current Control Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x03, &RegisterReading);
+    usleep(10);
+    LE_DEBUG(" Pre-Charge/Termination Current Control Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x04, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Charge Voltage Control Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x05, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Charge Termination/Timer Control Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x06, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Thermal Regulation Control Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x07, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Misc Operation Control Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x08, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("System Status Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x09, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Fault Register value is %d\n",RegisterReading);
+
+    SmbusReadReg(0, 0x6B, 0x0A, &RegisterReading);
+    usleep(10);
+    LE_DEBUG("Vender / Part / Revision Status Register value is %d\n",RegisterReading);
+}
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * This function changes battery charger output voltage and reads back register
+ *
+ */
+//-------------------------------------------------------------------------------------------------
+static void OutputBatterychargerVoltage
+(
+    void
+)
+{
+    uint8_t OutputVoltage;
+    SmbusWriteReg(0,0x6B,0x04,0xba);
```

Comment:
Be consistent with in using lower or uppercase for A-F values in hex literals.  My preference is `0xABCDEF01` because the contrast in size between the lowercase 'x' and the uppercase 'A' makes it easier to identify where the numeric part begins.  Also, be consistent in applying spaces after commas.